src/Native/System.Security.Cryptography.Native/pal_ssl.cpp

```diff
@@ -404,7 +414,45 @@ static HashAlgorithmType MapHashAlgorithmType(const SSL_CIPHER* cipher)
     return HashAlgorithmType::None;
 }
 
-extern "C" int32_t GetSslConnectionInfo(SSL* ssl, CipherAlgorithmType* dataCipherAlg, ExchangeAlgorithmType* keyExchangeAlg, HashAlgorithmType* dataHashAlg, int32_t* dataKeySize)
+static int32_t GetHashKeySize(const SSL_CIPHER* cipher)
```

Comment:
It may make sense to combine this method with MapHashAlgorithmType to cut down on the duplicated #if and switch code. The new method would have 2 "out" parameters.